docs/design/AnoncredsW3CCompatibility.md

```diff
@@ -0,0 +1,286 @@
+
+We also have to consider which classes or anoncreds objects have been modified
+
+The classes modified according to the same [PR](https://github.com/hyperledger/anoncreds-rs/pull/273) mentioned above are:<br>
+
+[Credential](https://github.com/hyperledger/anoncreds-rs/pull/273/files#diff-6f8cbd34bbd373240b6af81f159177023c05b074b63c7757fc6b3796a66ee240R402)<br>
+added class methods (`from_w3c`)<br>
+added instance methods (`to_w3c`)<br>
+added bindings functions (`credential_from_w3c`, `credential_to_w3c`)<br>
+
+[PresentCredential](https://github.com/hyperledger/anoncreds-rs/pull/273/files#diff-6f8cbd34bbd373240b6af81f159177023c05b074b63c7757fc6b3796a66ee240R603)<br>
+modified instance methods (`_get_entry`, `add_attributes`, `add_predicates`)<br>
+<br>
+
+### How will they fit into ACA-PY?
+
+There are two scenarios to consider when we want to add w3c format support to ACA-PY.
+
+- Creating a W3C VC credential from credential definition, and issuing and presenting it as is
+- Converting an already issued legacy anoncreds to W3C format(vice versa) so the converted credential can be issued or presented.
+
+#### Creating a W3C VC credential from credential definition, and issuing and presenting it as is
+
+The issuance, presentation and verification of legacy anoncreds are implemented in this [./aries_cloudagent/anoncreds](https://github.com/hyperledger/aries-cloudagent-python/tree/main/aries_cloudagent/anoncreds) directory. Therefore, we will also start from there.<br>
+
+Let us navigate these implementation examples through the respective processes of the concerning agents - **Issuer** and **Holder** as described in https://github.com/hyperledger/anoncreds-rs/blob/main/README.md.
+
+Looking at the [issuer.py](https://github.com/hyperledger/aries-cloudagent-python/blob/main/aries_cloudagent/anoncreds/issuer.py) file and this code block:
+
+```
+async def create_credential_offer(self, credential_definition_id: str) -> str:
+...
+...
+  credential_offer = CredentialOffer.create(
+                  schema_id or cred_def.schema_id,
+                  credential_definition_id,
+                  key_proof.raw_value,
+              )
+...
+```
+
+we can implement the same thing in w3c VC format to send a w3c credential offer like so:
+
+- W3C Credential Offer
+
+**NOTE: In the W3C VCDM, there is no concept of a credential offer, and most implementations of W3C VCs have no step where a credential is offered before it is issued. This is because, unlike AnonCreds, W3C VCs require no cryptographic commitment from the holder. So an alternative approach to credential offers is to simply not provide them in the W3C case. However, we think that it is simpler to expose the step as part of our support, thus giving the option for the future and making the correspondence between W3C and AnonCreds features more regular.**
```

Comment:
An AnonCreds credential, whether in legacy or W3C VC Format **must** have an Offer and Request, so absolutely must be supported.  It is necessary for the AnonCreds cryptography.  This will (by definition) limit the holders that can use "flexcred" credentials to those that can support AnonCreds, but it is our required starting point.

A constraint of what we are doing is that every AnonCreds credential can be issued in W3C VC Format, but not every W3C VC can be issued in AnonCreds format.  Notable limitation are complex JSON structures (although in theory, some could be flattened to work in AnonCreds) and JSON arrays or arbitrary length objects.  The latter cannot be supported in AnonCreds, since the CredDef in AnonCreds defines exactly how many items are in the issued credentials.

Reply:
Duly noted!